sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -74,6 +74,39 @@
 """,
 ).strip()
 
+# Query to fetch idle blocking sessions
+# with only basic sess and conn info
+# not joining with sys.dm_exec_requests
+IDLE_BLOCKING_SESSIONS_QUERY = re.sub(
+    r'\s+',
+    ' ',
+    """\
+SELECT
+    CONVERT(
+        NVARCHAR, TODATETIMEOFFSET(CURRENT_TIMESTAMP, DATEPART(TZOFFSET, SYSDATETIMEOFFSET())), 126
+    ) as now,
+    sess.login_name as user_name,
+    sess.last_request_start_time as last_request_start_time,
+    sess.session_id as id,
+    DB_NAME(sess.database_id) as database_name,
+    sess.status as session_status,
+    lqt.text as statement_text,
+    SUBSTRING(lqt.text, 1, {proc_char_limit}) as text,
+    c.client_tcp_port as client_port,
+    c.client_net_address as client_address,
+    sess.host_name as host_name,
+    sess.program_name as program_name
+FROM sys.dm_exec_sessions sess
+    INNER JOIN sys.dm_exec_connections c
+        ON sess.session_id = c.session_id
+    CROSS APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) lqt
```

Comment:
`dm_exec_connections. most_recent_sql_handle` is used to extract the sql statement for the idle session. As idle sessions do not exist in `dm_exec_requests`, we won't be able to the sql statement from `dm_exec_connections.sql_handle`